### PR TITLE
media-gfx/sam2p: Security bump

### PR DESCRIPTION
**Pull request by ajakk**
* EAPI bump
* sam2p-0.49.3-perl526.patch is obsoleted, drop it
* The necessity of sam2p-0.49.3-build-fixes.patch is dubious and it doesn't apply, drop it
* 0.49.4 fixes CVE-2017-{14628,14629,14630,14631,14636,14637} (Bug 631636)
* Patches included to fixes CVE-2018-{7552,7553,7554} (Bug 649750), and CVE-2018-16663 (Bug 636876)

Bug: https://bugs.gentoo.org/631636
Bug: https://bugs.gentoo.org/636876
Bug: https://bugs.gentoo.org/649750
Package-Manager: Portage-2.3.101, Repoman-2.3.22
Signed-off-by: John Helmert III <jchelmert3@posteo.net>

Currently I'm getting a "Pre-stripped files found" QA notice with it, but I currently don't see the solution to fixing it.

### CODE REVIEW

**Comment by ajakk** [comment]
Getting a different QA notice now with USE=examples, but it is not a regression, it happens with 0.49.3 too:

 * QA Notice: Colliding files found by ecompress:
 *
 *   /var/tmp/portage/media-gfx/sam2p-0.49.4/image/usr/share/doc/sam2p-0.49.4/examples/pts2.pbm
 *   /var/tmp/portage/media-gfx/sam2p-0.49.4/image/usr/share/doc/sam2p-0.49.4/examples/pts2.pbm.gz
 *   /var/tmp/portage/media-gfx/sam2p-0.49.4/image/usr/share/doc/sam2p-0.49.4/examples/fisht.pnm
 *   /var/tmp/portage/media-gfx/sam2p-0.49.4/image/usr/share/doc/sam2p-0.49.4/examples/fisht.pnm.gz

File: media-gfx/sam2p/files/sam2p-configure-strip.patch [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ ifeq ($(ENABLE_DEBUG), no)
+ CXXX=$(CXD_no)
+ endif
+@@ -101,8 +102,8 @@ IDE_MODES := release debug
+ IDE_TARGETS_debug := $(patsubst %,%.yes, $(TARGETS))
+ IDE_TARGETS_release := $(TARGETS)
+ 
+-CXD_assert =$(CXX) -s -O2
+-CXD_no     =$(CXX) -s -DNDEBUG -O3
++CXD_assert =$(CXX) $(GCC_STRIP) -O2
```
Comment (SoapGentoo) [inline media-gfx/sam2p/files/sam2p-configure-strip.patch:26]: doesn't this imply you'll be forcing `-O2` down users' throats regardless?

File: media-gfx/sam2p/sam2p-0.49.4.ebuild [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+src_compile() {
+	emake GCC_STRIP=
+}
+
+src_install() {
+	dobin sam2p
+	einstalldocs
+
+	if use examples; then
+		dodoc -r examples
```
Comment (SoapGentoo) [inline media-gfx/sam2p/sam2p-0.49.4.ebuild:57]: how large is this dir?

Comment (ajakk) [reply media-gfx/sam2p/sam2p-0.49.4.ebuild]: It's at 2972kb on my system.

File: media-gfx/sam2p/files/sam2p-configure-strip.patch (hunk 1 above)
Comment (ajakk) [reply media-gfx/sam2p/files/sam2p-configure-strip.patch]: Added a fix to the patch, seems like user-configurability of CXXFLAGS was broken since 0.49.3.

**Comment by ajakk** [comment]
Added a snapshot ebuild due to numerous security problems in the commit log since last release that don't have CVE's. All the patches mentioned previously are all upstreamed patches and are thus included in the snapshot.

File: media-gfx/sam2p/sam2p-0.49.4_p20190718.ebuild [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+EAPI=7
+
+inherit autotools toolchain-funcs
+
+COMMIT="b953f63307c4a83fa4615a4863e3fb250205cd98"
+
+DESCRIPTION="Utility to convert raster images to EPS, PDF and many others"
+HOMEPAGE="https://github.com/pts/sam2p"
+SRC_URI="https://github.com/pts/sam2p/archive/${COMMIT}.tar.gz -> ${P}.tar.gz"
+LICENSE="GPL-2"
```
Comment (SoapGentoo) [inline media-gfx/sam2p/sam2p-0.49.4_p20190718.ebuild:13]: empty line between `SRC_URI` and `LICENSE`

File: media-gfx/sam2p/sam2p-0.49.4_p20190718.ebuild [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sparc ~x86 ~x64-macos"
+IUSE="examples gif"
+
+DEPEND="dev-lang/perl"
```
Comment (SoapGentoo) [inline media-gfx/sam2p/sam2p-0.49.4_p20190718.ebuild:18]: `DEPEND` only on perl? :thinking: My guess is this should be in `BDEPEND`

File: media-gfx/sam2p/sam2p-0.49.4_p20190718.ebuild [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+RESTRICT="test"
```
Comment (SoapGentoo) [inline media-gfx/sam2p/sam2p-0.49.4_p20190718.ebuild:20]: put this right after `IUSE`

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**